### PR TITLE
Mvi missing xml prolog

### PR DESCRIPTION
**Pull request by kreek**
The MVI development environment doesn't return an XML prolog in the response, staging does 😕 
Updated the specs to include the prolog and added a prolog to the response if it is missing so that parsing is consistent.

### CODE REVIEW

**Comment by kreek** [comment]
@omgitsbillryan ah ok didn't know prolog wasn't required in 1.0. Ox is not throwing an error but what it considers the root node changes so the choice was to add the prolog if missing or have code that handles a different root.

File: lib/mvi/responses/base.rb [hunk 1]
```diff
@@ -43,6 +44,11 @@ def body
 
       private
 
+      def ensure_xml_prolog(xml)
+        xml.prepend('<?xml version="1.0" encoding="UTF-8"?>') unless xml =~ /^<\?xml/
```
Comment (kreek) [inline lib/mvi/responses/base.rb]: 1. The response that did include a prolog had the encoding set to UTF-8 so ya was assuming that was the case in the other responses.
2. Yes response.body is always XML

**Comment by omgitsbillryan** [comment]
@kreek oh, yuck.  So you chose the lesser of the two evils

**Comment by omgitsbillryan** [comment]
Whether you decide to take action or not based on my one comment...

LGTM!

**Comment by kreek** [comment]
👍 you should be able to xpath to a specific node anywhere in the doc though I'll try that first and if not merge this.

**Pull request merged**